x-pack/plugin/eql/src/main/java/org/elasticsearch/xpack/eql/action/EqlSearchResponse.java

```diff
@@ -116,6 +128,19 @@ public EqlSearchResponse(StreamInput in) throws IOException {
         asyncExecutionId = in.readOptionalString();
         isPartial = in.readBoolean();
         isRunning = in.readBoolean();
+        if (in.getTransportVersion().onOrAfter(TransportVersions.EQL_ALLOW_PARTIAL_SEARCH_RESULTS)) {
+            int size = in.readVInt();
+            if (size == 0) {
+                shardFailures = ShardSearchFailure.EMPTY_ARRAY;
+            } else {
+                shardFailures = new ShardSearchFailure[size];
+                for (int i = 0; i < shardFailures.length; i++) {
+                    shardFailures[i] = readShardSearchFailure(in);
+                }
```

Comment:
could we have used readArray here or some of the primitives? that probably saves some lines of code. Same for the write side of it.

Reply:
Done